app/assets/src/components/AlignmentViz.jsx

```diff
@@ -0,0 +1,276 @@
+import axios from "axios";
+import React from "react";
+
+class ReadViz extends React.Component {
+  constructor(props) {
+    super(props);
+    this.parsedAlignment = this.parseAlignment(props);
+  }
+
+  parseAlignment(readInfo) {
+    let readPart = parseInt(readInfo.name.split("/")[1])
```

Comment:
suggest destructuring `readInfo` like so:

```tsx
const { name, sequence, metrics, refInfo } = readInfo
```

and use the consts in the rest of the function.

(I checked that this syntax is available for `babel-preset-env, which IDseq has. So definitely use away!)

https://babeljs.io/docs/plugins/preset-env/

Reply:
I am actually changing some content in the readInfo object so can't quite do const.

Reply:
Ah, javascript `const` only signals that the value cannot be reassigned.

So,

```js
const timmy = { hairColor: 'blue', eyeColor: 'brown' }

timmy.hairColor = 'purple' // good. You can change the content within the object.

timmy = { hairColor: 'purple', eyeColor: 'brown' } // bad. Expected output: TypeError: invalid assignment to const `timmy'. Because we can't reassign timmy to a different object!
```